File: leetcode/python/0434_number_of_segments_in_a_string.py
class Solution:
    def countSegments(self, s: str) -> int:
        if len(s) == 0: return 0

        counting, res = False, 0

        for i in range(0, len(s)):
            if counting == False and s[i] != " ":
                counting = True
            elif counting == True and (s[i] == " " or i == len(s) - 1):
                counting = False
                res += 1

        if counting == True: res += 1

        return res

File: leetcode/java/0434_number_of_segments_in_a_string.java
class Solution {
    public int countSegments(String s) {
        if (s.length() == 1 && s.charAt(0) == ' ') return 0;
        if (s.length() == 1) return 1;

        boolean counting = false;
        int res = 0;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) != ' ' && !counting) {
                counting = true;
                if (i == s.length() - 1) res++;
            } else if ((s.charAt(i) == ' ' || i == s.length() - 1) && counting) {
                counting = false;
                res++;
            }
        }

        return res;
    }
}